### PR TITLE
Implemented lineage only mode

### LINKED ISSUE #51: Pangolin only mode

### PR DESCRIPTION
**Pull request by johausmann**
* Implementation of create reference only mode (Related to comment in #50) 
* Implementation of pangolin/lineage only mode to call pangolin for VCF or FASTA input. Closes #51

### CODE REVIEW

**Review by priesgo — CHANGES_REQUESTED** [review]
Would you please add tests to the CI that test for `reference_generate` and `lineage_mode`? 

Through automated tests you make sure that it works now and that if it's broken later on we know it before it's merged.

File: main.nf [hunk 1]
```diff
@@ -240,119 +246,149 @@ if (params.skip_bcftools && params.skip_gatk && params.skip_ivar && params.skip_
     exit 1
 }
 
+// Lineage mode should only be used when running with assembly or VCF input
+// For FASTQ input we can run pangolin as regular rule
+if (params.lineage_mode && input_fastqs) {
```
[the remark this answers is not in the document]
Comment (johausmann) [reply main.nf:251]: That's true, but don't the files from --fastq1 and --fastq2 also end up in `input_fastqs`? Depending on the input, isn't the created channel set to `input_fastqs` with ` .set{ input_fastqs}` ?

**Comment by johausmann** [comment]
Okay, I've addressed the issues. Let me know if anything else is missing

**Review by priesgo — APPROVED** [review]
Perfect!

**Pull request merged**